qstash/workflows/basics/api.mdx

```diff
@@ -0,0 +1,119 @@
+---
+title: "API"
+---
+
+# context.run
+
+`context.run` requires a step name and the code of the step as an async function. Each step can run for the full duration allowed by the platform. The result of each step is stored and remembered by QStash, allowing it to continue from where it left off in case of failure (e.g., endpoint downtime).
+
+Here are a couple of examples to demonstrate its usage:
+
+<CodeGroup>
+```javascript serial
+export const POST = serve<string>({
+  routeFunction: async (context) => {
+    const input = context.requestPayload;
+    const result1 = await context.run("step1", async () => {
+      return someWork(input);
+    });
+
+    await context.run("step2", async () => {
+      someOtherWork(result1);
+    });
+  },
+});
+```
+
+```javascript parallel
+export const POST = serve<string>({
+  routeFunction: async (context) => {
+    const input = context.requestPayload;
+    const promise1 = context.run("step1", async () => {
+      return someWork(input);
+    });
+
+    const promise2 = context.run("step2", async () => {
+      return someOtherWork(input);
+    });
+
+    await Promise.all([promise1, promise2]);
+  },
+});
+```
+</CodeGroup>
+
+# context.sleep
+
+`context.sleep` postpones the execution of the code for a specified duration. The code will exit and not consume any function time. After the specified duration, the endpoint will be called again to resume execution.
+
+The sleep duration can be as long as a week or a year, depending on your [QStash plan(Max Delay)](https://upstash.com/pricing/qstash).
+
+```javascript
+export const POST = serve<string>({
+  routeFunction: async (context) => {
+    const input = context.requestPayload;
+
+    const user = await context.run("signin", async () => {
+      const user = saveUserToDB(input);
+      return user;
+    });
+
+    await context.sleep("waitForWelcomeEmail", daysToSeconds(3));
+
+    await context.run("sendWelcomeEmail", async () => {
+      return sendEmail(user.name, user.email);
+    });
+  },
+});
+```
+
+# context.sleepUntil
+
+`context.sleepUntil` postpones the execution of the code until a specified date. The code will exit and not consume any function time. On the given date, the endpoint will be called again to resume execution.
```

Comment:
Is it worth mentioning that they need to `await` sleep calls? Otherwise, their flow might not work as expected.

A user might naively assume that execution will pause automatically (I'm not sure though). WDYT?